### DISCUSSION TITLE
Best practice for dividing/multiplying hidden channels?

### DISCUSSION THREAD

**Discussion by Boehm92** [pyg-team/pytorch_geometric#6927] [Q&A] [answered]
Hi all,

i have a question regarding the hidden_channels of a GCN. I have the example from the pytorch geometric website for graph classification:
```
class GCN(torch.nn.Module):
    def __init__(self, hidden_channels):
        super(GCN, self).__init__()
        torch.manual_seed(12345)
        self.conv1 = GCNConv(dataset.num_node_features, hidden_channels)
        self.conv2 = GCNConv(hidden_channels, hidden_channels)
        self.conv3 = GCNConv(hidden_channels, hidden_channels)
        self.lin = Linear(hidden_channels, dataset.num_classes)
```

Would'n I normally divide the hidden channels in each layer like following? 

```
class GCN(torch.nn.Module):
    def __init__(self, hidden_channels):
        super(GCN, self).__init__()
        torch.manual_seed(12345)
        self.conv1 = GCNConv(dataset.num_node_features, hidden_channels)
        self.conv2 = GCNConv(hidden_channels, hidden_channels  / 2)
        self.conv3 = GCNConv(hidden_channels / 2, hidden_channels / 4)
        self.lin = Linear(hidden_channels / 4, dataset.num_classes)
```

I checked the example for the FeaStConv layer, because it solves a similiar problem to mine, and they multiply the hidden channels. Can someone explain what the best practice for GNNs would be and why?

Kind Regards
Stefan

**Comment by rusty1s** [comment, marked as answer]
Ofc, you can do this, but this is a hyperparameter that you can tune for a certain model/dataset combination. I don't think it is a very important setting to tune.

**Reply by Boehm92** [reply]
thanks a lot, then i stcick to the example models
